src/main/java/edu/kpi/testcourse/storage/UrlRepositoryFakeImpl.java

```diff
@@ -25,10 +27,30 @@ public void createUrlAlias(UrlAlias urlAlias) {
     return aliases.get(alias);
   }
 
+  /**
+   * Delete the UrlAlias if exists from aliases HashMap.
+   *
+   * @param email of the user to whom the UrlAlias belongs
+   * @param alias the UrlAlias which should be removed
+   *
+   * @throws RuntimeException if no such email or alias
+   * @throws edu.kpi.testcourse.storage.UrlRepository.PermissionDenied
+   *         if passed email is not equal to stored email in found UrlAlias object
+   */
+
   @Override
   public void deleteUrlAlias(String email, String alias) {
-    // TODO: We should implement it
-    throw new UnsupportedOperationException();
+    UrlAlias foundUrlAlias = findUrlAlias(alias);
+
+    if (foundUrlAlias == null || foundUrlAlias.email() == null) {
```

Comment:
Redundant if statement part (foundUrlAlias.email() == null)
```suggestion
    if (foundUrlAlias == null) {
         throw new RuntimeException();
    }
```